returnn/training.py

```diff
@@ -177,10 +184,26 @@ def __init__(
             "time": time_rqmt,
         }
 
-        if self.use_horovod:
-            self.rqmt["cpu"] *= self.horovod_num_processes
-            self.rqmt["gpu"] *= self.horovod_num_processes
-            self.rqmt["mem"] *= self.horovod_num_processes
+        if self.pe_num_processes:
+            assert self.horovod_num_processes
+            assert self.horovod_num_processes >= self.pe_num_processes
+            assert self.horovod_num_processes % self.pe_num_processes == 0
+        if (self.horovod_num_processes or 1) > (self.pe_num_processes or 1):
+            assert self.horovod_num_processes % (self.pe_num_processes or 1) == 0
+            self.rqmt["cpu"] *= self.horovod_num_processes // (
+                self.pe_num_processes or 1
+            )
+            self.rqmt["gpu"] *= self.horovod_num_processes // (
+                self.pe_num_processes or 1
+            )
+            self.rqmt["mem"] *= self.horovod_num_processes // (
+                self.pe_num_processes or 1
+            )
+
+        if self.pe_num_processes:
+            self.rqmt.setdefault("qsub_args", []).extend(
```

Comment:
maybe we can add a new requirement category to sisyphus (e.g. rqmt["pe"]) and then each engine would implement it accordingly? 

@critias

Reply:
Yep, this part here is very SGE specific, which is bad. Other engines could maybe assert that no `qsub_args` are set in rqmt, but then it would only work with SGE.

Reply:
Done. https://github.com/rwth-i6/sisyphus/pull/114